packages/gatsby/cache-dir/ssr-develop-static-entry.js

```diff
@@ -111,7 +115,81 @@ export default (pagePath, isClientOnlyPage, callback) => {
 
     const pageData = getPageData(pagePath)
 
-    const componentChunkName = pageData?.componentChunkName
+    const { componentChunkName, staticQueryHashes = [] } = pageData
+
+    let scriptsAndStyles = flatten(
+      [`commons`].map(chunkKey => {
+        const fetchKey = `assetsByChunkName[${chunkKey}]`
+
+        let chunks = get(stats, fetchKey)
+        const namedChunkGroups = get(stats, `namedChunkGroups`)
+
+        if (!chunks) {
+          return null
+        }
+
+        chunks = chunks.map(chunk => {
+          if (chunk === `/`) {
+            return null
+          }
+          return { rel: `preload`, name: chunk }
+        })
+
+        namedChunkGroups[chunkKey].assets.forEach(asset =>
+          chunks.push({ rel: `preload`, name: asset })
+        )
+
+        const childAssets = namedChunkGroups[chunkKey].childAssets
+        for (const rel in childAssets) {
+          chunks = concat(
+            chunks,
+            childAssets[rel].map(chunk => {
+              return { rel, name: chunk }
+            })
+          )
+        }
+
+        return chunks
+      })
+    )
+      .filter(s => isObject(s))
+      .sort((s1, s2) => (s1.rel == `preload` ? -1 : 1)) // given priority to preload
+
+    scriptsAndStyles = uniqBy(scriptsAndStyles, item => item.name)
+
+    const styles = scriptsAndStyles.filter(
+      style => style.name && style.name.endsWith(`.css`)
+    )
+
+    styles
+      .slice(0)
+      .reverse()
+      .forEach(style => {
+        headComponents.unshift(
+          <script
+            dangerouslySetInnerHTML={{
+              __html: `
+            // Immediately update the styles link with its absolute URL so HMR
+            // will work
+            var scripts = document.getElementsByTagName('script')
+            var currentScript = scripts[scripts.length - 1]
+            var linkEl = currentScript.previousSibling
+            var newHref = new URL("${__PATH_PREFIX__}/${style.name}", location.href).href
+            linkEl.href = newHref
+            `,
+            }}
```

Comment:
This feels weird - is this needed for .css only? we don't do anything like that for `commons.js`?

Reply:
Would `data-href` attribute be a bit nicer? I'll try to play with it and see if we can avoid rewriting `href` like that, because it does seem quite hacky (and also with doing `previousSibling` trick, especially that users might be able to reorder things with `gatsby-ssr` APIs

Reply:
This is a (weird IMO) restriction of mini-css-extract-plugin — it only supports absolute links for HMR atm https://github.com/webpack-contrib/mini-css-extract-plugin/issues/407#issuecomment-568912471 — we could fix it upstream but who knows when we'd get to use it. So this seems like the best option in the meantime.

Reply:
Hmm, are you sure this is needed? I just checked code/tests of miniextractcssplugin and it seems to work without any shenanigans:

https://github.com/webpack-contrib/mini-css-extract-plugin/blob/master/test/HMR.test.js#L154-L179

https://github.com/webpack-contrib/mini-css-extract-plugin/blob/359c00fa7f02f2dc662c461ac7cc926a7b61c9bd/test/__snapshots__/HMR.test.js.snap#L23-L25

(it also works for me locally after I dropped this script)

Reply:
I guess magic of browsers make it work (I was on `google.com` and messed in browser console):
![Screenshot 2020-12-21 at 20 59 14](https://user-images.githubusercontent.com/419821/102817105-6cf1a880-43cf-11eb-81b2-84e647654fb4.png)